### PR TITLE
Enabled bpm service for tag based approval

### PR DESCRIPTION
**Pull request by hsong-rh**
This PR depends on https://github.com/RedHatInsights/approval-process/pull/22

### CODE REVIEW

File: spec/controllers/v1.0/requests_controller_spec.rb [hunk 1]
```diff
@@ -429,7 +429,7 @@
 
   # Test suite for GET /requests/:request_id/requests
   describe 'GET /requests/:request_id/requests' do
-    let(:parent_request) { create(:request, :name => "parent", :owner => "jdoe", :number_of_children => 2, :number_of_finished_children => 0, :tenant_id => tenant.id) }
+    let!(:parent_request) { create(:request, :name => "parent", :owner => "jdoe", :number_of_children => 2, :number_of_finished_children => 0, :tenant_id => tenant.id) }
```
Comment (bzwei) [inline spec/controllers/v1.0/requests_controller_spec.rb]: why the change is required? there is no other change in this spec

File: app/controllers/api/v1/stageaction_controller.rb [hunk 2]
```diff
@@ -73,15 +73,15 @@ def set_stage
...
 
       def set_order
-        request = @stage.request
+        request = @request
         {
           :orderer       => request.requester_name,
-          :product       => request.content["product"],
-          :portfolio     => request.content["portfolio"],
-          :order_id      => request.content["order_id"],
+          :product       => request.request_context.content["product"],
```
Comment (bzwei) [inline app/controllers/api/v1/stageaction_controller.rb]: you can still use `request.content`

File: app/controllers/api/v1/stageaction_controller.rb [hunk 3]
```diff
@@ -39,9 +39,9 @@ def update
           comments = params[:message]
         end
 
-        Insights::API::Common::Request.with_request(@stage.request.context.transform_keys(&:to_sym)) do
-          ActsAsTenant.with_tenant(Tenant.find(@stage.tenant_id)) do
-            ActionCreateService.new(@stage.id).create(
+        Insights::API::Common::Request.with_request(@request.request_context.context.transform_keys(&:to_sym)) do
```
Comment (bzwei) [inline app/controllers/api/v1/stageaction_controller.rb]: you can access `@request.context`

**Comment by bzwei** [comment]
can we rename `stageaction_controller` to `requestaction_controller`?

File: spec/controllers/v1.0/requests_controller_spec.rb (hunk 1 above)
Comment (hsong-rh) [reply spec/controllers/v1.0/requests_controller_spec.rb]: This change fixed the previous spec test failure.

**Comment by hsong-rh** [comment]
I feel it's better to keep it as `stageaction_controller`, in sync with the endpoint `/api/v1.0/stageaction`, which is specialized for us.
